packages/medusa-plugin-algolia/src/services/algolia.js

```diff
@@ -0,0 +1,141 @@
+    this.client_ = algoliasearch(application_id, admin_api_key)
+  }
+
+  /**
+   * Add two numbers.
+   * @param {string} indexName - The name of the index
+   * @param {*} options - not required just to match the schema we are used it
+   * @return {*}
+   */
+  createIndex(indexName, options = {}) {
+    return this.client_.initIndex(indexName)
+  }
+
+  /**
+   * Used to get an index
+   * @param {string} indexName  - the index name.
+   * @return {Promise<{object}>} - returns response from search engine provider
+   */
+  getIndex(indexName) {
+    let hits = []
+    return this.client_
+      .initIndex(indexName)
+      .browseObjects({
+        query: indexName,
+        batch: (batch) => {
+          hits = hits.concat(batch)
+        },
+      })
+      .then(() => hits)
+  }
+
+  /**
+   *
+   * @param {string} indexName
+   * @param {Array} documents - products list array
+   * @param {*} type
+   * @return {*}
+   */
+  addDocuments(indexName, documents, type) {
+    const transformedDocuments = this.getTransformedDocuments(type, documents)
+    return this.client_.initIndex(indexName).saveObjects(transformedDocuments)
+  }
+
+  /**
+   * Used to replace documents
+   * @param {string} indexName  - the index name.
+   * @param {Object} documents  - array of document objects that will replace existing documents
+   * @param {Array.<Object>} type  - type of documents to be replaced (e.g: products, regions, orders, etc)
+   * @return {Promise<{object}>} - returns response from search engine provider
+   */
+  replaceDocuments(indexName, documents, type) {
+    const transformedDocuments = this.getTransformedDocuments(type, documents)
+    return this.client_
+      .initIndex(indexName)
+      .replaceAllObjects(transformedDocuments)
+  }
+
+  /**
+   * Used to delete document
+   * @param {string} indexName  - the index name
+   * @param {string} document_id  - the id of the document
+   * @return {Promise<{object}>} - returns response from search engine provider
+   */
+  deleteDocument(indexName, document_id) {
+    return this.client_.initIndex(indexName).deleteObject(document_id)
+  }
+
+  /**
+   * Used to delete all documents
+   * @param {string} indexName  - the index name
+   * @return {Promise<{object}>} - returns response from search engine provider
+   */
+  deleteAllDocuments(indexName) {
+    return this.client_.initIndex(indexName).delete()
+  }
+
+  /**
+   * Used to search for a document in an index
+   * @param {string} indexName - the index name
+   * @param {string} query  - the search query
+   * @param {*} options
+   * - any options passed to the request object other than the query and indexName
+   * - additionalOptions contain any provider specific options
+   * @return {*} - returns response from search engine provider
+   */
+  search(indexName, query, options) {
+    if (options.additionalOptions) {
+      options = { ...options, ...options.additionalOptions }
+      delete options.additionalOptions
+    }
```

Comment:
This will remove additional and pagination options from the query to Algolia. 

Instead we should transform `options` here into something, that Algolia can use.
```
// this is what you receive in the method here
options = { paginationOptions, filter, additionalOptions }
```
```
// this is what Algolia can consume
index.search(string query, {
  searchParameters: { }
  requestOptions: { }
})
```
So we need to transform `options` into `searchParameters` and `requestOptions`.

More on these properties and they can hold [here](https://www.algolia.com/doc/api-reference/api-methods/search/#method-param-requestoptions).

Does this make sense?

Reply:
got it will do it